app/views/facility_journals/_problem_transactions.html.haml

```diff
@@ -0,0 +1,68 @@
+          - if @extra_date_column
+            %th.nowrap= OrderDetail.human_attribute_name(@extra_date_column_header || @extra_date_column)
+          - if current_facility.blank? || cross_facility?
+            %th= Facility.model_name.human
+          %th.nowrap= [Account.model_name.human, Account.human_attribute_name("expires_at")].join(" ")
+          %th.order-note= OrderDetail.human_attribute_name("description")
+          %th.nowrap= OrderDetail.human_attribute_name("user")
+          - unless @account
+            %th= Account.model_name.human
+            %th= Account.human_attribute_name("owner")
+          %th.currency= OrderDetail.human_attribute_name("cost")
+          %th= OrderDetail.human_attribute_name("order_status")
+          - if local_assigns[:show_statements]
+            %th= Statement.model_name.human
+      %tbody
+        - order_details.each do |order_detail|
+          %tr{ class: row_class(order_detail) }
+            - if @order_detail_link
+              %td.nowrap= link_to @order_detail_link[:text], @order_detail_link[:proc].call(order_detail) if @order_detail_link[:display?].call(order_detail)
+            - if backend?
+              %td.nowrap= link_to order_detail.order.id, facility_order_path(order_detail.order.facility, order_detail.order)
+              %td.nowrap= link_to order_detail.id, manage_order_detail_path(order_detail), class: 'manage-order-detail'
+            - else
+              %td.nowrap= link_to order_detail.order.id, order_path(order_detail.order)
+              %td.nowrap= link_to order_detail.id, order_order_detail_path(order_detail.order, order_detail)
+            %td= order_detail.send(:"#{@date_range_field}").try(:strftime, "%m/%d/%Y")
+            - if @extra_date_column && order_detail.send(@extra_date_column)
+              %td= order_detail.send(@extra_date_column).strftime("%m/%d/%Y")
+            - if current_facility.blank? || cross_facility?
+
+              %td= order_detail.order.facility
+            %td= order_detail.account.expires_at.try(:strftime, "%m/%d/%Y")
+            %td.user-order-detail.order-note
+              .order-detail-description
+                = order_detail_description(order_detail)
+                - order_detail.extend PriceDisplayment
+                = "(#{order_detail.wrapped_quantity})".html_safe unless order_detail.problem?
+                - if order_detail.time_data.present?
+                  %br
+                  %em= order_detail.time_data
+              - if order_detail.note.present?
+                .order-detail-extra.order-detail-note
+                  = render "shared/order_detail_note", order_detail: order_detail
+
+
+            %td= order_detail.order.user.full_name
+            - unless @account
+              %td= order_detail.account
+              %td= order_detail.account.owner_user
+            %td.currency
+              - order_detail.send(:extend, PriceDisplayment)
```

Comment:
Possibly a decorator object that returns the cells or the table row that might be easier to refactor?